core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -120,8 +121,10 @@ public FileOutStream createFile(AlluxioURI path, CreateFileOptions options)
     URIStatus status;
     try {
       masterClient.createFile(path, options);
-      status = masterClient.getStatus(path, GetStatusOptions.defaults().setLoadMetadataType(
-          LoadMetadataType.Never));
+      // Do not sync before this getStatus, since the UFS file is expected to not exist.
+      status = masterClient.getStatus(path,
+          GetStatusOptions.defaults().setLoadMetadataType(LoadMetadataType.Never)
+              .setCommonOptions(CommonOptions.defaults().setSyncIntervalMs(-1)));
```

Comment:
Is this necessary, since the file should be incomplete, it will short-circuit in the `syncMetadata` logic?

Reply:
You are right, the incomplete file is already taken care of. I added this just as a safeguard, probably for the same reason that load metadata was also set to `NEVER`.

Reply:
Got it, thanks.